### DISCUSSION TITLE
Why does Mantine only support px (and not rem/em)

### DISCUSSION THREAD

**Discussion by baba43** [mantinedev/mantine#1965] [Feedback]
Currently Mantine only supports pixel units, which is also pointed out in the documentation. I would like to know if there is a reason or even a necessity for this? 

For my understanding relative units offer some advantages and are more flexible in many cases. That's why we find them in most modern projects (that I know). But maybe I'm overlooking the disadvantages, which I'm happy to be enlightened about.

In any case, I'd be happy to talk about them. Maybe we can also talk about where we might be able to use alternative units after all, and where we definitely shouldn't in order to remain compatible.

If this discussion already exists somewhere, I'm happy with a link. I haven't found anything off the top of my head.

Thanks for the attention and this great project.

**Comment by rtivital** [comment]
Hi @baba43, we do not provide support for other units because all Mantine components are built with `px`. It would be a hard job to support other units and will require a lot of additional time to maintain this feature. `px` units work well in all modern browsers, and I do not think that migrating the project to other units will worth the effort.

**Comment by heycassidy** [comment]
For font-size in particular, it's important to be able to use rems. Font sizes using px aren't affected by the browser's default font size setting, which is an accessibility issue.

**Reply by 7iomka** [reply]
> For font-size in particular, it's important to be able to use rems. Font sizes using px aren't affected by the browser's default font size setting, which is an accessibility issue.

In projects which don't use css-in-js but only css/scss styles for this case I prefer to keep all units in px but use some tool to convert px to rem in final build.
Like `postcss-pxtorem`
But for css in js I don't know if exists some similar tools.
I read about `stylis` and it plugins, but it hard to make some working plugin (In my case I tried to do plugin analogue for `gap-polyfill` but I still don't understand how to do it)

**Comment by kahboom** [comment]
Just wanted to add to @heycassidy 's comment. Relative units are particularly important for WCAG compliance, since many users rely on zooming. Here's an [article about it](https://www.24a11y.com/2019/pixels-vs-relative-units-in-css-why-its-still-a-big-deal/).

I really like Mantine but I've heard other devs bring up the topic around it not supporting other units, saying it's one of the biggest reasons they can't/don't use it. Are there any plans for this in the future, even if it's just for the font size? Thanks!
